stress/common/src/main/java/alluxio/stress/CachingBlockMasterClient.java

```diff
@@ -0,0 +1,41 @@
+package alluxio.stress;
+
+
+import alluxio.grpc.LocationBlockIdListEntry;
+import alluxio.master.MasterClientContext;
+import alluxio.worker.block.BlockMasterClient;
+import alluxio.worker.block.BlockStoreLocation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Use this class to avoid map-list conversion at the client side
+ * So if you are running all processes on the same machine,
+ * the client side work affect the performance less.
+ * */
+public class CachingBlockMasterClient extends BlockMasterClient {
```

Comment:
In `BlockMasterClient.heartbeart()` and `BlockMasterClient.register()`, the worker calls `convertBlockListMapToProto` to convert its block data structure into a proto. When we are doing a stress test (using simulated workers), this happens in every simulated RPC. I try to avoid this because now the RTT for that RPC includes a non-trivial client-side work and this noise can get very big if we are testing with more than millions of blocks.

I called this `Caching` because it "caches" the converted proto. Yea the name is not that good... I'll give it more thoughts

Reply:
I feel like controlling `convertBlockListMapToProto` is better than probably mocking `register` and `heartbeat` methods, as the latter logic are more subject to change.